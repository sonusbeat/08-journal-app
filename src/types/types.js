const types = {
  // Autentication
  login: "[Auth] Login",
  logout: "[Auth] Logout",

  // Errors
  uiSetError: "[UI] Set Error",
  uiRemoveError: "[UI] Remove Error",

  // Loading State
  uiStartLoading: "[UI] Start loading",
  uiFinishLoading: "[UI] Finish loading",

  // Notes
  notesAddNew:         "[Notes] Note Added",
  notesActive:         "[Notes] Note Activated",
  notesLoad:           "[Notes] Entries Loaded",
  notesUpdated:        "[Notes] Updated note",
  notesFileUrl:        "[Notes] Updated image url",
  notesDelete:         "[Notes] Deleted note",
  notesLogoutCleaning: "[Notes] Logged out cleaned",

};

export default types;