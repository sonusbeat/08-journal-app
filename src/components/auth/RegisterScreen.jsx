import { useDispatch } from "react-redux";
import { Link } from "react-router-dom";
import validator from "validator";
import useForm from '../../hooks/useForm';
import { setError, unsetError } from '../../actions/ui';

const RegisterScreen = () => {
  const dispatch = useDispatch();

  const [ formValues, handleInputChange ] = useForm({
    name: "Daniel",
    email: "sonusbeat@gmail.com",
    password: "0123456789",
    password_confirmation: "0123456789",
  });

  const { name, email, password, password_confirmation } = formValues;

  const isFormValid = () => {
    if ( validator.isEmpty( name ) ) {
      dispatch( setError("Name should be present !") );
      return false;
    } else if ( name.trim().length <= 2) {
      dispatch( setError("Name should be greater than 2 characters !") );
      return false;
    } else if ( validator.isEmpty( email ) ) {
      dispatch( setError("Email should be present !") );
      return false;
    } else  if ( !validator.isEmail( email ) ) {
      dispatch( setError("Email format not valid !") );
      return false;
    } else  if ( validator.isEmpty( password ) ) {
      dispatch( setError("Password should be present !") );
      return false;
    } else  if ( password.trim().length < 8 ) {
      dispatch( setError("Password should be greater than 8 characters !") );
      return false;
    } else  if ( validator.isEmpty( password_confirmation ) ) {
      dispatch( setError("Password confirmation should be present !") );
      return false;
    } else  if ( password_confirmation.trim().length < 8 ) {
      dispatch( setError("Password confirmation should be greater than 8 characters !") );
      return false;
    } else  if ( password !== password_confirmation ) {
      dispatch( setError("Password and Password confirmation should be match each other !") );
      return false;
    }

    dispatch( unsetError() );

    return true;
  };

  const handleRegister = (e) => {
    e.preventDefault();

    if ( isFormValid() ) {
      console.log("Se envia el formulario");
    }
  };

  return (
    <>
      <div className="auth__alert-error mb-2">
        Mensaje de Error
      </div>

      <h3 className="auth__title">Register</h3>

      <form onSubmit={ handleRegister }>

        <input
          name="name"
          value={ name }
          className="auth__input"
          type="text"
          placeholder="name"
          autoComplete="off"
          onChange={ handleInputChange }
        />

        <input
          name="email"
          value={ email }
          className="auth__input"
          type="text"
          placeholder="email"
          autoComplete="off"
          onChange={ handleInputChange }
        />

        <input
          type="password"
          value={ password }
          className="auth__input"
          placeholder="password"
          name="password"
          autoComplete="off"
          onChange={ handleInputChange }
        />

        <input
          name="password_confirmation"
          value={ password_confirmation }
          className="auth__input"
          type="password"
          placeholder="Confirm password"
          autoComplete="off"
          onChange={ handleInputChange }
        />

        <button
          className="btn btn-primary btn-block"
        >Register</button>

          <Link className="auth__link link" to="/auth/login">
            Already Registered ?
          </Link>
      </form>
    </>
  );
}

export default RegisterScreen;