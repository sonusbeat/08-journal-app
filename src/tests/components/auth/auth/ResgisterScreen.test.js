import { mount } from "enzyme";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

import RegisterScreen from "../../../../components/auth/RegisterScreen";
import types from '../../../../types/types';

// jest.mock("../../../../actions/auth", () => ({
  // Simular algo aqui
// }));

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

const initialState = {
  auth: {},
  ui: {
    loading: false,
    msgError: null
  }
};

let store = mockStore( initialState );

// Simular el dispatch para las pruebas
// store.dispatch = jest.fn();

const wrapper = mount(
  <Provider store={ store }>
    <MemoryRouter>
      <RegisterScreen />
    </MemoryRouter>
  </Provider>
);

describe("Pruebas en <RegisterScreen />", () => {

  // beforeEach(() => {
    // Limpiar el store volviendolo a sus valores iniciales
    // store = mockStore( initialState );

    // Buena practica limpiar todos los Mocks
    // jest.clearAllMocks();
  // });

  test("Deberia de coincidir con el Snapshot", () => {

    expect( wrapper ).toMatchSnapshot();

  });

  test("Debe de hacer el dispatch de la acciÃ³n respectiva", () => {

    const nameField = wrapper.find( 'input[name="name"]');

    nameField.simulate("change", {
      target: {
        value: "Daniel",
        name: "name"
      }
    });

    // Disparar el formulario con simulate
    wrapper.find("form").simulate("submit", {
      preventDefault(){}
    });

    const actions = store.getActions();

    expect( actions[0] ).toEqual({
      type: types.uiSetError,
      payload: "Email should be present !"
    });

  });

});