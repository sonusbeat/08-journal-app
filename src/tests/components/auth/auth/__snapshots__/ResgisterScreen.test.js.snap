// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RegisterScreen /> Deberia de coincidir con el Snapshot 1`] = `
Array [
  <h3
    className="auth__title"
  >
    Register
  </h3>,
  <form
    className="animate__animated animate__fadeIn animate__faster"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="name"
      type="text"
      value=""
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="email"
      type="text"
      value=""
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="password"
      value=""
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="password_confirmation"
      onChange={[Function]}
      placeholder="Confirm password"
      type="password"
      value=""
    />
    <button
      className="btn btn-primary btn-block"
    >
      Register
    </button>
    <a
      className="auth__link link"
      href="/auth/login"
      onClick={[Function]}
    >
      Already Registered ?
    </a>
  </form>,
]
`;
